src/dune_rules/install_rules.ml

```diff
@@ -815,19 +798,51 @@ let packages_file_is_part_of path =
     let+ map = packages sctx in
     Option.value (Path.Build.Map.find map path) ~default:Package.Id.Set.empty
 
-let install_rules sctx (package : Package.t) =
+module Sctx_and_package = struct
+  module Super_context = Super_context.As_memo_key
+
+  type t = Super_context.t * Package.t
+
+  let hash (x, y) = Hashtbl.hash (Super_context.hash x, Package.hash y)
+
+  let equal (x1, y1) (x2, y2) = x1 == x2 && y1 == y2
```

Comment:
Why not use structural equality for packages?

Reply:
Probably for performances.